src/hotspot/os_cpu/linux_x86/os_linux_x86.cpp

```diff
@@ -459,22 +459,47 @@ bool os::supports_sse() {
 }
 
 juint os::cpu_microcode_revision() {
+  // Note: this code runs on startup, and therefore should not be slow.
+
   juint result = 0;
-  char data[2048] = {0}; // lines should fit in 2K buf
+  char data[128] = {0}; // looking for short "microcode" line, should be enough
   size_t len = sizeof(data);
-  FILE *fp = os::fopen("/proc/cpuinfo", "r");
-  if (fp) {
-    while (!feof(fp)) {
+
+  // Attempt 1 (faster): Read the microcode version off the sysfs.
+  {
+    FILE *fp = os::fopen("/sys/devices/system/cpu/cpu0/microcode/version", "r");
+    if (fp) {
       if (fgets(data, len, fp)) {
-        if (strstr(data, "microcode") != NULL) {
-          char* rev = strchr(data, ':');
-          if (rev != NULL) sscanf(rev + 1, "%x", &result);
-          break;
+        sscanf(data, "%x", &result);
+      }
+      fclose(fp);
+    }
+    if (result != 0) {
+      return result;
+    }
+  }
+
+  // Attempt 2 (slower): Read the microcode version off the procfs.
+  {
+    FILE *fp = os::fopen("/proc/cpuinfo", "r");
```

Comment:
Should this now slow-path continue to use a 2048 element buffer?

Reply:
No, because AFAICS, `fgets` reads until buffer is full or the EOL is read. We are looking for a rather short "microcode: 0x..." line in procfs, so there is no need to allocate the entire 2K for this. This was my first attempt to improve this code, and it is not directly relevant to issue at hand, so I can revert it, if that's a questionable change.

Reply:
Ok, makes sense. If it's safe and works I don't mind if you keep it. I guess JDK-8283200 will make such micro-optimizations moot, though.

Reply:
Did it anyway, see new commit.